### ISSUE TITLE
Running ncu against a pnpm project with pnpm-workspace.yaml that has overrides crashes with unhandled rejection.

### ISSUE THREAD

**Issue by ara1353-dev** [raineorshine/npm-check-updates#2055]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the [latest version](https://www.npmjs.com/package/npm-check-updates?activeTab=versions) of `npm-check-updates`
- [x] I am using a [compatible version of node](https://github.com/raineorshine/npm-check-updates/blob/main/package.json#L20)

---

Regression introduced by #1767 (v22.2.6) added new call sites for `isPackageManagerProtocol(version)` without guarding against version being undefined, which occurs when ncu resolves packages from workspace overrides that are not direct dependencies.

## Steps to Reproduce

- create a pnpm project with `"packageManager": "pnpm"`
- Add a `pnpm-workspace.yaml` with overrides pointing to transitive (non-direct) dependencies:
```
overrides:
    "@typescript-eslint/utils": "8.56.0"
    "diff: "8.0.3"
```

Ru: ncu --target latest

## Current Behavior

TypeError: Cannot read properties of undefined (reading 'startsWith')
       at isPackageManagerProtocol  (index.js:4718)
Unhandled Rejection! This is a bug and should be reported: https://github.com/raineorshine/npm-check-updates/issues

## Expected Behavior

ncu completes successfully
